Thussein mechanical bc

tarekhussein opened the pull request:
- The equations of linear elasticity are implemented in dependence on displacements, without interpretation in the total strain equations.
- The displacements have not exact values yet, I suggest the problem might be in functions _setup_solver_eps(self) and _solve_u(self), where the linear solver is implemented @pzimbrod

tarekhussein committed:
implement 2D mesh

tarekhussein committed:
Implement mechanical BC
- the Linear elasticity equations are done

tarekhussein committed:
Implement mechanical BC

tarekhussein committed:
Implement mechanical BC

tarekhussein committed:
Implement mechanical BC

pzimbrod committed:
Extend README and add Pip environment (#12)

* add requirements.txt

* extend README.md

pzimbrod reviewed (changes requested):
I haven't validated results in Paraview yet, but I made some comments on what I noticed on the code itself, it's mostly minor and naming convention things.

pzimbrod reviewed (changes requested):
I found the culprit for the model to not update

tarekhussein commented on main.py:
```diff
@@ -36,7 +38,7 @@
     "T_ambient": 600.0,
     # Initial temperature
-    "T_0": 800.0,
+    "T_0": 923.1,
```
The article of J.H. Nielson mentioned the intial temperture as 923.1 K, Although, this value might differ in our Surroglas project.

tarekhussein committed:
Update mechanical_BC
- exchange Functions names' "displacements" into "U"
- apply expression & interpolate to get mechanical strain values.

tarekhussein commented on ThermoViscoProblem.py:
```diff
@@ -344,7 +358,61 @@ def _setup_solver(self) -> None:
+        self.a = inner(elastic_sigma(self.u_trial), elastic_epsilon(self.v_test)) * dx 
+        self.L = dot(self.ss, self.v_test) * dx + dot(self.traction,self.v_test) * ds
+
+        self.expressions = {}
```
I tried to write and recall this expression function from ViscoelasticModel.py, but elastic_epsilon() could not be accessed as it is function inside a function.

pzimbrod replied:
Yeah, I get the problem. Maybe it makes sense to refactor `elastic_sigma` and `elastic_epsilon` into `ViscoelasticModel`. We can then access all functions and the expression via `self.material_model.elastic_sigma`.

Another thing I noticed: you set the key for the expression in line 408 to `"U"`, but isn't that the mechanical (elastic) strain? If so, it would make more sense to create a key `"elastic_strain"` since we called the thermal strain `"thermal_strain"` within the expressions dictionary:
https://github.com/PI-UniA/SurroGlas/blob/7356bd8183be36dd5c7b1187c1fb8519a1a78734/ViscoelasticModel.py#L127-L133

tarekhussein committed:
Refactor Elastic stresses and strains
- rename mech_strain into elastic_strain
-simplify the 2D mesh

tarekhussein committed:
Exchange np.isclose to fem.locate_dofs_topological
- optimize lame's parameter

tarekhussein committed:
Use geo kernel instead of occ kernel

tarekhussein committed:
Update mechanical BC
- the model is able to deal with different forms of mesh and BCs

tarekhussein committed:
Implement "if" condition for scaled time
- to avoid convergence into zero value, a conditional case is applied

pzimbrod committed:
refactor BC markers into dict

pzimbrod committed:
only enforce bottom BC
